NebulaHost/PacketProcessors/Factory/CreatePrebuildsRequestProcessor.cs

```diff
@@ -0,0 +1,42 @@
+﻿using NebulaModel.Attributes;
+using NebulaModel.Networking;
+using NebulaModel.Packets.Factory;
+using NebulaModel.Packets.Processors;
+using NebulaWorld.Factory;
+using HarmonyLib;
+using System.Collections.Generic;
+
+namespace NebulaHost.PacketProcessors.Factory
+{
+    [RegisterPacketProcessor]
+    class CreatePrebuildsRequestProcessor : IPacketProcessor<CreatePrebuildsRequest>
+    {
+        public void ProcessPacket(CreatePrebuildsRequest packet, NebulaConnection conn)
+        {
+            PlayerAction_Build pab = GameMain.mainPlayer.controller?.actionBuild;
+            if (pab != null) {
+                //Make backup of values that are overwritten
+                List<BuildPreview> tmpList = pab.buildPreviews;
+                bool tmpConfirm = pab.waitConfirm;
+                UnityEngine.Vector3 tmpPos = pab.previewPose.position;
+                UnityEngine.Quaternion tmpRot = pab.previewPose.rotation;
+
+                //Create Prebuilds from incomming packet
+                pab.buildPreviews = packet.GetBuildPreviews();
+                pab.waitConfirm = true;
+                FactoryManager.EventFromServer = true;
```

Comment:
This should probably be an `FactoryManager.EventFromClient` since we are in the host code here.

Reply:
I'll fix it in main